model_zoo/vision/lenet2.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+import argparse
+import os.path
+import lbann
+
+# ----------------------------------
+# Command-line arguments
+# ----------------------------------
+
+desc = ('Construct and run LeNet on MNIST data. '
+        'Running the experiment is only supported on LC systems.')
+parser = argparse.ArgumentParser(description=desc)
+parser.add_argument(
+    '--partition', action='store', type=str,
+    help='scheduler partition', metavar='NAME')
+parser.add_argument(
+    '--account', action='store', type=str,
+    help='scheduler account', metavar='NAME')
+args = parser.parse_args()
+
+# ----------------------------------
+# Construct layer graph
+# ----------------------------------
+
+# Input data
+input = lbann.Input()
+images = lbann.Identity(input)
+labels = lbann.Identity(input)
+
+# LeNet
+x = lbann.Convolution(images,
+                      num_dims = 2,
+                      num_output_channels = 6,
+                      num_groups = 1,
+                      conv_dims_i = 5,
+                      conv_strides_i = 1,
+                      conv_dilations_i = 1,
+                      has_bias = True)
+x = lbann.Relu(x)
+x = lbann.Pooling(x,
+                  num_dims = 2,
+                  pool_dims_i = 2,
+                  pool_strides_i = 2,
+                  pool_mode = "max")
+x = lbann.Convolution(x,
+                      num_dims = 2,
+                      num_output_channels = 16,
+                      num_groups = 1,
+                      conv_dims_i = 5,
+                      conv_strides_i = 1,
+                      conv_dilations_i = 1,
+                      has_bias = True)
+x = lbann.Relu(x)
+x = lbann.Pooling(x,
+                  num_dims = 2,
+                  pool_dims_i = 2,
+                  pool_strides_i = 2,
+                  pool_mode = "max")
+x = lbann.FullyConnected(x, num_neurons = 120, has_bias = True)
+x = lbann.Relu(x)
+x = lbann.FullyConnected(x, num_neurons = 84, has_bias = True)
+x = lbann.Relu(x)
+x = lbann.FullyConnected(x, num_neurons = 10, has_bias = True)
+probs = lbann.Softmax(x)
+
+# Loss function and accuracy
+loss = lbann.CrossEntropy([probs, labels])
+acc = lbann.CategoricalAccuracy([probs, labels])
+
+# ----------------------------------
+# Setup experiment
+# ----------------------------------
+
+# Setup model
+mini_batch_size = 64
+num_epochs = 4
+model = lbann.Model(mini_batch_size,
+                    num_epochs,
+                    layers=lbann.traverse_layer_graph(input),
+                    objective_function=loss,
+                    metrics=[lbann.Metric(acc, name='accuracy', unit='%')],
+                    callbacks=[lbann.CallbackPrint(), lbann.CallbackTimer()])
+
+# Setup optimizer
+opt = lbann.SGD(learn_rate=0.01, momentum=0.9)
+
+# Setup DataReader
+message = lbann.lbann_pb2.DataReader()
```

Comment:
This won't work in the latest `develop`. The `DataReader` message is now in `reader.proto`/`reader_pb2.py`, which is not formally imported into `lbann`.